crates/rspack_plugin_limit_chunk_count/src/lib.rs

```diff
@@ -0,0 +1,582 @@
+mod chunk_combination;
+
+use std::{
+  cmp::Ordering,
+  collections::{HashMap, HashSet},
+};
+
+use chunk_combination::{ChunkCombination, ChunkCombinationBucket, ChunkCombinationUkey};
+use rspack_core::{
+  BoxModule, Chunk, ChunkGraph, ChunkGroup, ChunkUkey, ModuleGraph, OptimizeChunksArgs, Plugin,
+  PluginContext, PluginOptimizeChunksOutput, RuntimeSpec,
+};
+use rspack_database::Database;
+use rspack_util::comparators::compare_ids;
+
+// TODO: we should remove this function to crate rspack_util
+fn compare_modules_by_identifier(a: &BoxModule, b: &BoxModule) -> std::cmp::Ordering {
```

Comment:
if you want you can move this to `rspack_core::util` since it depends `BoxModule`, we probably will re-organize `rspack_util` in the near future

Reply:
This is a bit complex because this method relies on `BoxModule`, which causes a module circular reference error in `rspack_util`.